packages/core/src/VSelection.ts

```diff
@@ -0,0 +1,179 @@
+     * `select(a, RelativePosition.BEFORE, b, RelativePosition.AFTER)` => `[ab]`
+     * `select(a, RelativePosition.AFTER, b, RelativePosition.AFTER)` => `a[b]`
+     *
+     * @param anchorNode
+     * @param [anchorPosition] default: `RelativePosition.BEFORE`
+     * @param [focusNode] default: `startNode`
+     * @param [focusPosition] default: `RelativePosition.AFTER`
+     */
+    select(anchorNode: VNode, focusNode?: VNode): this;
+    select(
+        anchorNode: VNode,
+        anchorPosition: RelativePosition,
+        focusNode: VNode,
+        focusPosition: RelativePosition,
+    ): this;
+    select(
+        anchorNode: VNode,
+        anchorPosition: RelativePosition | VNode = RelativePosition.BEFORE,
+        focusNode: VNode = anchorNode,
+        focusPosition: RelativePosition = RelativePosition.AFTER,
+    ): this {
+        if (anchorPosition instanceof VNode) {
+            focusNode = anchorPosition;
+            anchorPosition = RelativePosition.BEFORE;
+        }
+        if (focusPosition === RelativePosition.AFTER) {
+            this.setFocus(focusNode, focusPosition);
+            this.setAnchor(anchorNode, anchorPosition);
+        } else {
+            this.setAnchor(anchorNode, anchorPosition);
+            this.setFocus(focusNode, focusPosition);
+        }
+        return this;
+    }
+    /**
+     * Set the anchor of the selection by targetting a `reference` VNode and
+     * specifying the `position` in reference to that VNode ('BEFORE', 'AFTER'),
+     * like in an `xpath`. If no relative position if given, include the
+     * reference node in the selection. Return self.
+     *
+     * @param reference
+     * @param [position]
+     */
+    setAnchor(reference: VNode, position = RelativePosition.BEFORE): this {
+        if (this.direction === Direction.FORWARD) {
+            this.range.setStart(reference, position);
+        } else {
+            this.range.setEnd(reference, position);
+        }
+        return this;
+    }
+    /**
+     * Set the focus of the selection by targetting a `reference` VNode and
+     * specifying the `position` in reference to that VNode ('BEFORE', 'AFTER'),
+     * like in an `xpath`. If no relative position if given, include the
+     * reference node in the selection. Return self.
+     *
+     * @param reference
+     * @param [position]
+     */
+    setFocus(reference: VNode, position = RelativePosition.AFTER): this {
+        if (this.direction === Direction.FORWARD) {
+            this.range.setEnd(reference, position);
+        } else {
+            this.range.setStart(reference, position);
+        }
+        return this;
+    }
+    /**
+     * Extend the selection from its anchor to the given location, targetting a
+     * `reference` VNode and specifying the `direction` of the extension.
+     *
+     * @param reference
+     * @param [direction] default: Direction.FORWARD
+     */
+    extendTo(reference: VNode, direction = Direction.FORWARD): this {
```

Comment:
So this basically duplicates the logic of `range.extendTo` but with the possibility of giving it a direction?

Reply:
It does not duplicate the logic of `range.extendTo`. The range one is moving either start or end depending on which one is closer to the node to include. This one moves the focus of the selection such that it selects the given node. Example:
```
<p>ab[cd]ef</p>
```
Calling `range.extendTo(b)` will produce:
```
<p>a[bcd]ef</p>
```
While `selection.extendTo(b)` will produce:
```
<p>a]b[cdef</p>
```
Basically `selection.extendTo` is the equivalent of the DOM's `selection.extend`, while the range one is "smarter" in the way that it trully *extends* the range, conserving what was *already* in it.

Reply:
Ok, got it. That's a bit confusing thought, I think it might require some extra doc effort.

Reply:
When we will use `extendTo` ?
There is no test for these methods and no code calls them